### PR TITLE
api(cdc): drain capture return 503 if not all captures initialized

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: ref #4757 

### What is changed and how it works?

When call the open api to drain the capture, all captures should be initialized, otherwise return 503 indicate that the client should retry.

#### Tests 

 - Unit test

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: cdc/owner/owner.go [hunk 1]
```diff
@@ -410,21 +410,36 @@ func (o *ownerImpl) clusterVersionConsistent(captures map[model.CaptureID]*model
...
-	changefeedWithTableCount := 0
-	totalTableCount := 0
+	var (
+		changefeedWithTableCount int
+		totalTableCount          int
+		err                      error
+	)
 	for _, changefeed := range o.changefeeds {
-		count := changefeed.scheduler.DrainCapture(query.CaptureID)
+		count, e := changefeed.scheduler.DrainCapture(query.CaptureID)
```
Comment (overvenus) [inline cdc/owner/owner.go:419]: Please check error right after `DrainCapture`

```suggestion
		count, e := changefeed.scheduler.DrainCapture(query.CaptureID)
		if e != nil {
			err = e
			break
		}
```

File: cdc/scheduler/internal/v3/coordinator.go [hunk 2]
```diff
@@ -139,59 +140,61 @@ func (c *coordinator) Rebalance() {
...
+func (c *coordinator) DrainCapture(target model.CaptureID) (int, error) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
+	if !c.captureM.CheckAllCaptureInitialized() {
+		log.Info("tpscheduler: drain capture request ignored, "+
+			"since not all captures initialized",
+			zap.String("target", target))
+		// return false to let client retry.
+		return 0, cerror.ErrSchedulerRequestFailed.
```
Comment (overvenus) [inline cdc/scheduler/internal/v3/coordinator.go:153]: Please add a test to cover this line.

**Review by overvenus — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**